### ISSUE TITLE
Add ability to display empty Overlay in DynamicMap

### ISSUE THREAD

**Issue by philippjfr** [pyviz/holoviews#3620]
Now that we support dynamically adding new subplots we should also allow displaying a completely empty (Nd)Overlay and populating it with plots after the initial render.

**Comment by jlstevens** [comment]
Sound like this is related to https://github.com/pyviz/holoviews/issues/3696

**Comment by TheoMathurin** [comment]
Is there now a way to return an empty (Nd)Overlay? Currently returning either with an empty list/dictionary yields an error.

In general there are workarounds (_e.g._ returning 0-size or transparent elements) but there aren't ideal.

**Comment by jlstevens** [comment]
> Is there now a way to return an empty (Nd)Overlay

I don't believe this has been addressed yet though I do agree this would be very desirable.

Given't Philipp's comment above, it sounds like this should now be possible so I will assign this to the next milestone.
